source/session.ts

```diff
@@ -625,10 +625,8 @@ export class Session {
 
       if (value != null && typeof value.valueOf() === "string") {
         value = `"${value}"`;
-      } else if (value && value._isAMomentObject) {
-        // Server does not store microsecond or timezone currently so
-        // need to strip from query.
-        value = moment(value).utc().format(ENCODE_DATETIME_FORMAT);
+      } else if (convertToISOString(value)) {
+        value = convertToISOString(value);
```

Comment:
Is the comment not true?

>  // Server does not store microsecond or timezone currently so
> // need to strip from query.

These will be translated to a SQL comparison, so I think we need to strip to match how the dates are stored in the database.

Reply:
it's true that the server currently does not store microseconds, but I believe it's better to allow sending in all ISO8691 validated strings and convert them to whatever is supported in the server (as it currently works), than removing microseconds in the client.

server does store timezones in a way by only supporting UTC timezone, and we are thus converting to UTC dates always.

Reply:
I tried testing this manually, and it does look like it works, though I am not sure why.. both of these return the expected Note.

```
select id from Note where thread_activity is "2023-02-20 09:58:01"
select id from Note where thread_activity is "2023-02-20T09:58:01.102Z"
```

Having a date-times as `identifyingKey` doesn't make much sense, but I assume we should continue to support it. As long as it does, I am happy.

Reply:
the server encode all dates to a precision the database handles, which means that it strips the milliseconds (and timezone) from the string, which is why it is backwards compatible.

if the server API one day start supporting milliseconds granularity, this query will stop working yes, but that would be a breaking change from the server API anyway.